.github/workflows/sbom.yml

```diff
@@ -0,0 +1,34 @@
+name: Release workflow for creating SBOM and attaching to the release
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  sbom:
+    name: Generate and upload SBOM
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+    strategy:
+      matrix:
+        # build and publish in parallel: linux/386, linux/amd64, windows/386, windows/amd64, darwin/amd64
+        goos: [linux, darwin]
+        goarch: [amd64, arm64]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Generate SBOM
+        uses: CycloneDX/gh-gomod-generate-sbom@v2
+        with:
+          version: v1
+          args: mod -licenses -output ${{ matrix.goos }}-${{ matrix.goarch }}.bom.xml
```

Comment:
Please note that Dependencytrack unfortunately does not understand the evidence format for licenses.

I found that the `--assert-licenses` option is required to add to cyclonedx for use with Dependencytrack. This results in the license being asserted instead of being listed in an evidence section.